stockroom/storages/modelstore.py

```diff
@@ -5,93 +5,130 @@
         for i, w in enumerate(weights):
-            asetn = parser.model_asetkey_from_details(aset_prefix, dtypes[i])
-            aset = get_aset(co, asetn, dtypes[i], longest, variable=True)
-            aset[i] = w.reshape(-1)
+            modelKey = parser.modelkey(name, longest, dtypes[i])
+            model_aset = self.get_arrayset(co, modelKey)
+            if model_aset is None:
+                model_aset = self.create_arrayset(
+                    co, modelKey, int(longest), dtypes[i], variable_shape=True)
+            model_aset[i] = w.reshape(-1)
             if w.shape:
                 shape_aset[i] = np.array(w.shape)
             else:
                 shape_aset[i] = np.array(()).astype('int64')
+        self.close_arraysets()
         co.close()
 
     def load(self, name, model):
-        import torch
         root = get_stock_root()
         head_commit = get_current_head(root)
         co = self.repo.checkout(commit=head_commit)
-        aset_prefix = co.metadata[parser.metakey(name, 'aset_prefix')]
-        dtypes = parser.string_to_dtypes(co.metadata[parser.metakey(name, 'dtypes')])
-        library = co.metadata[parser.metakey(name, 'library')]
-        num_layers = int(co.metadata[parser.metakey(name, 'num_layers')])
+        library, longest, dtypes, numLayers, layers = self.get_metadata(co, name)
+        shapeKey = parser.shapekey(name, longest)
+        shape_aset = self.get_arrayset(co, shapeKey)
         weights = []
-        for i in range(num_layers):
-            asetn = parser.model_asetkey_from_details(aset_prefix, dtypes[i])
-            aset = get_aset(co, asetn)
-            shape_asetn = parser.shape_asetkey_from_model_asetkey(name)
-            shape_aset = co.arraysets[shape_asetn]
+        for i in range(numLayers):
+            modelKey = parser.modelkey(name, longest, dtypes[i])
+            aset = self.get_arrayset(co, modelKey)
             w = aset[i].reshape(np.array(shape_aset[i]))
             weights.append(w)
-        if len(weights) != num_layers:
-            raise RuntimeError("Critical: length doesn't match. Raise an issue")
+        if len(weights) != numLayers:
+            raise RuntimeError(
+                "Critical: Discrepancy in the stored data: numLayers != len(weights)")
         if library == 'torch':
-            str_layers = co.metadata[parser.metakey(name, 'layers')]
-            layers = parser.string_to_layers(str_layers)
-            if len(layers) != num_layers:
-                raise RuntimeError("Critical: length doesn't match. Raise an issue")
-            state = {layers[i]: torch.from_numpy(weights[i]) for i in range(num_layers)}
+            import torch
+            state = {layers[i]: torch.from_numpy(weights[i]) for i in range(numLayers)}
             model.load_state_dict(state)
         else:
             model.set_weights(weights)
 
+    @staticmethod
+    def store_metadata(co, name, library, longest, dtypes, numLayers, layers=None):
```

Comment:
Why static method? Isn't the checkout a persistent variable? (I really don't know?)

Also, is this a method which should be called by the user? it seems like it's an internal implementation detail? if so, prepend the method name with an underscore (`_`)